opendevin/memory/condenser.py

```diff
@@ -1,26 +1,199 @@
+from typing import Callable
+
 from opendevin.core.logger import opendevin_logger as logger
+from opendevin.core.utils import json
 from opendevin.llm.llm import LLM
 
+MAX_TOKEN_COUNT_PADDING = 1024
```

Comment:
Any reason this is increased from 512 to 1024?

Reply:
Three small reasons: 😄 
- action prompt for monologue contains task and background commands, and a couple of other things that are not referenced here but still used. The task in particular is not sent, but it can be... anything. I've seen people comprehensively specify a whole project
- litellm uses two tokenizers iirc, hugging face and openai. It defaults any other model to openai tokenizer. I have no idea yet if that is appropriate or not, if we're risking or not. So just in case...
- I've had some weird cases recently, after we introduced tokens counting, when monologue failed at the first steps with ContextWindowExceeded, for GPT-3.5. I thought we could try a bit harder to preempt it.

Reply:
This is changing:
- I now see the last reason as being for a lower value, not a higher value, as long as some reasonable estimation
- we're going back to the catching the exception first, and only do it preemptively if the user set a config max tokens.